lib/iris/fileformats/netcdf.py

```diff
@@ -1865,47 +1893,32 @@ def _create_cf_data_variable(self, cube, dimension_names, local_keys=None,
             The newly created CF-netCDF data variable.
 
         """
-        if 'packing' in kwargs:
-            packing = kwargs.pop('packing')
-            if packing:
-                scale_factor = None
-                add_offset = None
-                fill_value = None
-                if isinstance(packing, dict):
-                    if 'dtype' not in packing:
-                        msg = "The dtype attribute is required for packing."
-                        raise ValueError(msg)
-                    dtype = np.dtype(packing['dtype'])
-                    if 'scale_factor' in packing:
-                        scale_factor = packing['scale_factor']
-                    if 'add_offset' in packing:
-                        add_offset = packing['add_offset']
-                    if 'fill_value' in packing:
-                        fill_value = packing['fill_value']
+
+        if packing:
+            if isinstance(packing, dict):
+                if 'dtype' not in packing:
+                    msg = "The dtype attribute is required for packing."
+                    raise ValueError(msg)
+                dtype = np.dtype(packing['dtype'])
+                scale_factor = packing.get('scale_factor', None)
+                add_offset = packing.get('add_offset', None)
+            else:
+                masked = ma.isMaskedArray(cube.data)
```

Comment:
This would trigger the data to be loaded. Is that what you intended?

Reply:
Sorry not your code... Perhaps you wouldn't mind adding a comment though:
```
# We compute the scale_factor based on the min/max of the data. This requires the data to be loaded.
```

Reply:
> This would trigger the data to be loaded. Is that what you intended?
I hadn't noticed this, it looks like it comes from the old code. It can be fixed, but it would require getting the min and max of the data while storing it. I can do that if you like, but it might be better to do it as a separate PR

Reply:
No, just adding the comment is good.

Reply:
Doesn't the docstring at [this comment](https://github.com/SciTools/iris/pull/2747/files/5d8375a2cae41dd7041073f9e472fade533dc1ae#r135280518) cover it ?

Reply:
Not really. It is about making the code readable. Doesn't need to be an essay.